drv/stm32xx-i2c/src/lib.rs

```diff
@@ -148,33 +159,48 @@ pub enum ReadLength {
     Variable,
 }
 
+#[allow(clippy::upper_case_acronyms)]
 #[derive(Copy, Clone, Eq, PartialEq)]
```

Comment:
we may want to derive `Count` for this as well, and then add `#[count(children)]` to the events that include a `Register`?
```suggestion
#[derive(Copy, Clone, Eq, PartialEq, counters::Count)]
```

Reply:
Deliberate to not do this; see my other comments.